### PR TITLE
Propagate Kimi accuracy-compatible flag to PaddleFleet provider

### PR DESCRIPTION
**Pull request by zrr1999**
## Summary
- when KimiK2Provider receives use_accuracy_compatible=True, also set the PaddleFleet internal gpt_model_use_experimental_version flag
- keeps the public PaddleFormers knob as the single user-facing accuracy gate

## Validation
- python -m py_compile paddleformers/transformers/kimi_k2/modeling.py
- used in the Kimi reduced-checkpoint alignment workspace together with PaddleFleet accuracy-compatible paths

### CODE REVIEW

File: paddleformers/transformers/kimi_k2/modeling.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ class KimiK2Provider(GPTModelProvider):
 
     def __post_init__(config):
         super().__post_init__()
+        if getattr(config, "use_accuracy_compatible", False):
+            # PaddleFleet still uses this internal flag for several
+            # precision-alignment branches. Keep the public PaddleFormers
+            # knob as the single user-facing gate.
+            config.gpt_model_use_experimental_version = True
```
Comment (risemeup1111) [inline paddleformers/transformers/kimi_k2/modeling.py:45]: P1 **优先级：P1**

这里直接打开 `gpt_model_use_experimental_version` 会让 `use_accuracy_compatible=True` 的 Kimi-K2 在 `sequence_parallel=True` 时运行到 PaddleFleet 的断言：`GPTEmbedding.forward` 要求 experimental 版本下不能同时使用 `multi_latent_attention` 和 sequence parallel，而 Kimi-K2 在建 provider 前固定设置了 `multi_latent_attention=True`。这会把原本可配置的 TP/SP 训练路径变成首次 forward 才失败。请在映射内部标志前做显式兼容性校验（或补齐 PaddleFleet 支持），避免静默构造出必崩配置。

```suggestion
        if getattr(config, "use_accuracy_compatible", False):
            if getattr(config, "sequence_parallel", False) and getattr(config, "multi_latent_attention", False):
                raise ValueError(
                    "use_accuracy_compatible=True maps to PaddleFleet gpt_model_use_experimental_version, "
                    "which does not support Kimi-K2 MLA with sequence_parallel=True."
                )
            # PaddleFleet still uses this internal flag for several
            # precision-alignment branches. Keep the public PaddleFormers
            # knob as the single user-facing gate.
            config.gpt_model_use_experimental_version = True
```

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
发现一个需要合入前处理的兼容性问题，已在行级评论中给出具体原因和建议修改。CI 当前还有任务在运行中，未作为本次阻塞依据。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Comment by zrr1999** [comment]
Fixed in `640095a`: when `use_accuracy_compatible=True` is combined with `sequence_parallel=True` and Kimi MLA, `KimiK2Provider.__post_init__` now raises a `ValueError` before mapping to PaddleFleet `gpt_model_use_experimental_version`, instead of constructing a config that only fails later in forward.

**Review by risemeup1111 — APPROVED** [review]
复查了新提交，之前关于 `use_accuracy_compatible=True` 与 Kimi-K2 MLA/sequence parallel 组合的 P1 兼容性问题已经通过显式 `ValueError` 校验修复。当前代码改动对我来说可以接受。

CI 仍有任务在运行中，请以最终 CI 结果为准。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Comment by zrr1999** [comment]
## Kimi-K2 numerical-alignment scope update

This provider PR intentionally exposes one public `use_accuracy_compatible` gate and rejects the unsupported Kimi MLA + sequence-parallel combination early. The numerical results below belong to the complete gated PaddleFormers/PaddleFleet stack, not to this provider-only diff by itself.

Current validated Kimi-K2 2-layer BF16 scope:

- EP1 and EP8 forward loss: 100/100 steps bitwise exact.
- EP1 step-1 canonical parameter gradients: 794/794 `BITWISE_PASS`, including a rerun after PFCCLab/Megatron-LM was updated to `a10a87c5b`.
- EP8 bounded canonical gradients: 64/64 `BITWISE_PASS`.
- Bounded optimizer state: 8/8 exact for FP32 master parameter, FP32 moment1/moment2, and BF16 model copyback on two real parameters.

This does **not** yet establish complete native-training alignment. Global grad clipping, scheduler/warmup, complete parameter groups/tied weights, distributed optimizer semantics, and multi-step native updates are still pending.

The gate must also remain model/authority aware. The same author's PaddleFormers #4692 and the PFCCLab Megatron-LM/ms-swift alignment series target MiniMax/GLM4-MoE. Megatron-LM PR #2 pairs that stack with PyTorch-compatible AdamW and changes several Torch backward/training semantics. Kimi instead keeps native TransformerEngine 2.16.1 `FusedAdam` authoritative; forcing `torch.optim.AdamW` changed optimizer state hashes, whereas the TE-compatible Paddle path is 8/8 exact.

For that reason this PR should continue to provide only Kimi configuration propagation and capability validation. It should not imply that MiniMax-specific optimizer, MoE WGRAD/index-select backward, or per-token loss-normalization policies are universally valid for Kimi.
